napari/layers/surface/surface.py

```diff
@@ -282,14 +289,37 @@ def gamma(self, value):
     def _set_view_slice(self):
         """Sets the view given the indices to slice with."""
 
-        not_disp = list(self.dims.not_displayed)
-        disp = list(self.dims.displayed)
-        indices = np.array(self.dims.indices)
+        # Take vertex_values dimensionality into account if more than one value
+        # is provided per vertex.
+        if self.vertex_values.ndim > 1:
+            values = self.vertex_values[
+                self.dims.indices[: -self.vertices.shape[1]]
+            ]
+            if values.ndim > 1:
+                # If not all dimensions corresponding to the vertex_values
+                # are being sliced through, average over any remaining ones
+                # so that there is always only value per vertex.
+                values = values.mean(axis=tuple(range(0, values.ndim - 1)))
+            self._view_vertex_values = values
+            # Determine which axes of the vertices data are being displayed
+            # and not displayed, but ignoring the additional dimensions
+            # corresponding to the vertex_values.
+            indices = np.array(self.dims.indices[-self.vertices.shape[1] :])
+            additional_ndim = self.vertex_values.ndim - 1
+            disp = np.subtract(self.dims.displayed, additional_ndim)
+            disp = [d for d in disp if d >= 0]
+            not_disp = np.subtract(self.dims.not_displayed, additional_ndim)
+            not_disp = [d for d in not_disp if d >= 0]
+        else:
+            self._view_vertex_values = self.vertex_values
+            indices = np.array(self.dims.indices)
+            not_disp = list(self.dims.not_displayed)
+            disp = list(self.dims.displayed)
 
         self._data_view = self.vertices[:, disp]
         if len(self.vertices) == 0:
             self._view_faces = np.zeros((0, 3))
-        elif self.ndim > self.dims.ndisplay:
+        elif self.vertices.shape[1] > self.dims.ndisplay:
```

Comment:
As above: give a name to this quantity. spatial_ndim?

Reply:
done